### PR TITLE
Initial commit.

### PR DESCRIPTION
**Pull request by tonybaltovski**

### CODE REVIEW

File: clearpath_base/package.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  <depend>rclcpp</depend>
+  <depend>sensor_msgs</depend>
+  <depend>std_msgs</depend>
+  <depend>std_srvs</depend>
+  <depend>tf2</depend>
+  <depend>tf2_ros</depend>
+
+  <exec_depend>clearpath_control</exec_depend>
+  <exec_depend>clearpath_platform_description</exec_depend>
+  <exec_depend>micro_ros_agent</exec_depend>
```
Comment (roni-kreinin) [inline clearpath_base/package.xml:37]: Do we want to depend on micro_ros for platforms that don't use it?

**Comment by tonybaltovski** [comment]
@roni-kreinin renamed to `clearpath_platform`.

File: clearpath_platform/launch/platform.launch.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        PythonLaunchDescriptionSource(PathJoinSubstitution(
+        [FindPackageShare("clearpath_control"), 'launch', 'teleop_base.launch.py'])))
+
+    # Launch clearpath_control/teleop_joy.launch.py which is tele-operation using a physical joystick.
+    launch_clearpath_teleop_joy = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(PathJoinSubstitution(
+        [FindPackageShare("clearpath_control"), 'launch', 'teleop_joy.launch.py'])))
+
+    # Group for actions needed for the Clearpath Jackal J100 platform.
+    group_j100_action = GroupAction(
```
Comment (roni-kreinin) [inline clearpath_platform/launch/platform.launch.py:39]: I think we should have just a `group_platform_action` with all nodes in it. Then for nodes such as `micro_ros_agent` you can add a condition for  robot_model==j100

File: clearpath_platform/launch/platform.launch.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.substitutions import EnvironmentVariable, FindExecutable, LaunchConfiguration, PathJoinSubstitution
+from launch_ros.actions import Node
+from launch_ros.substitutions import FindPackageShare
+
+def generate_launch_description():
+    robot_model = LaunchConfiguration('robot_model')
+
+    # Launch Arguments
+    arg_robot_model = DeclareLaunchArgument(
```
Comment (roni-kreinin) [inline clearpath_platform/launch/platform.launch.py:13]: Should we call this arg `platform`?

**Review by roni-kreinin — APPROVED** [review]

**Pull request merged**